File: Codeforces/Problems/B_Ehab_Is_an_Odd_Person.cpp
#include <bits/stdc++.h>

using namespace std;

void solve()
{
    int n;
    cin >> n;

    vector<int> array(n);
    for (int i = 0; i < n; i++)
        cin >> array[i];

    for (int i = 0; i < n; i++)
    {
        bool swapped = false;
        for (int j = 0; j < n - i - 1; j++)
        {
            if (array[j] + array[j + 1] % 2 != 0)
            {
                if (array[j] > array[j + 1])
                {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    swapped = true;
                }
            }
        }
        if (!swapped) break;
    }

    for (auto &val: array)
        cout << val << " ";
}

int main()
{
    solve();
}

File: Codeforces/Problems/B_Ehab_Is_an_Odd_Person.py
def solve():
    n = int(input())

    array = list(map(int, input().split()))

    for i in range(n):
        swapped = False
        for j in range(n - i - 1):
            if array[j] + array[j + 1] % 2 != 0:
                if array[j] > array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]
                    swapped = True
        
        if not swapped:
            break
    
    print(*array)

if __name__ == '__main__':
    solve()